### DISCUSSION TITLE
Connect to Metamask wallet doesn't work on Mobile

### DISCUSSION THREAD

**Discussion by MarioCodetria** [WalletConnect/walletconnect-monorepo#2004] [Developer Support]
Connecting to my site works fine on PC, but on mobile (Android), when I press connect it opens a popup to select which app I want to use. I select Metamask and it opens the app, but doesn't display any popup requesting to accept connection with the site. I tried this in the official Web3Modal page and I can connect correctly. Is there something wrong with my config? The site is made in React.js.

I also noticed that on mobile, Metamask doesn't show up in the list of wallets when I open the site in PC mode.

I've seen a few related discussions, but none of them provided a definitive solution. Is Metamask not supported on mobile?

Versions:
"react": "^18.2.0",
"react-dom": "^18.2.0", 
"ethers": "^5.7.2",
"@web3modal/ethereum": "^2.1.1",
"@web3modal/html": "^2.1.1",
"@web3modal/react": "^2.1.1",
"wagmi": "^0.11.5",

Wallet Config:
const chains = [apiConstants.useDevelopmentChain ? goerli : mainnet];

// Wagmi client
```
const { provider } = configureChains(chains, [
    walletConnectProvider({
        projectId: '***',
    }),
]);
export const wagmiClient = createClient({
    autoConnect: true,
    connectors: modalConnectors({
        projectId: '***',
        version: '2',
        appName: 'web3Modal',
        chains,
    }),
    provider,
});

// Web3Modal Ethereum Client
export const ethereumClient = new EthereumClient(wagmiClient, chains);
```

App.tsx:
```
                    <WagmiConfig client={wagmiClient}>
                        {/* */}
                    </WagmiConfig>
                    <Web3Modal
                        projectId="***"
                        ethereumClient={ethereumClient}
                    />
```

**Comment by MarioCodetria** [comment]
Ok, apparently, V2 doesn't support Metamask. However, I tried changing "version" in modalConnectors to '1', but it didn't help. Metamask now shows up in the modal in mobile, but when I select open with Metamask, it either does nothing, or it says the session is already active, even tho I don't have any active sessions. How can I configure my project to work like the official Web3Modal website?

**Comment by paintoshi** [comment]
Having the same problem with @web3modal/react@2.3.7 using version 1, on iPhone. I can connect to metamask on mobile (when using chrome for example) but only works on the second try when clicking the "use backup" button. On the first try it opens app store instead for metamask.

But the problem is later when doing transactions from the web3modal app, it only opens the app store and not properly switching to the metamask app. Tested with Trust wallet and that works as expected.

**Comment by Defilearning** [comment]
Hi, just checking is there any solution/workaround for this?

**Comment by dnmintroad** [comment]
Also having the same exact issue. On iphone chrome and safari, connecting via metamask opens the app store. Proceeding to clicking "open" brings me to the app, but I don't get a prompt for a connection to the site. Going back to the brower to hit "retry" and "use backup" sometimes works, but it's not guaranteed.

Hopefully there's a solution for this

**Comment by bobwith2bees** [comment]
My Metamask notes / cross references: 

* V1 has an open issue https://github.com/MetaMask/metamask-mobile/issues/6387
* Wait for the Wallet Connect v2 branch to merge: https://github.com/MetaMask/metamask-mobile/tree/feat/walletconnect-v2

**Comment by acrite22** [comment]
Hello, I'm also have the same issue.  On Safari with MetaMask 6.3.0, connecting with MetaMask opens the app store.  Going back to Safari and trying again sometimes works.  My web app is using Vue3

**Comment by guiguy2018** [comment]
same issue with mobile chrome on ios iphone.  Even with the official web3modal test site, it sometimes work but mostly it can't connect to metamask.

**Comment by yenkhoon** [comment]
did you guys hit "Connection request declined" ?

**Comment by yenkhoon** [comment]
metamask have launched the released for supporting on walletconnect v2, is it the version of web3modal that im using causing issue? 
"@web3modal/ethereum": "^2.0.0-rc.3",
    "@web3modal/react": "^2.0.0-rc.3",

**Reply by guiguy2018** [reply]
Using
web3modal 2.6.2
Mobile Metamask IOS v7.2.0(1142)
Mobile Chrome IOS 114.0.5735.124
iPhone IOS 16.5.1

From Mobile phone chrome, https://web3modal.com, hit connect.  Then connect your wallet popup, select Metamak.  Metamask launches but the “pop up to connect to an address” does not appear 99% of the time.  

Scan qr code from another device metamask wallet does work.

I also tested with Uniswap and Trust wallets and they work much better.

**Reply by yenkhoon** [reply]
are you using react-scripts? to upgrade to version 2.6.2, typescript has to be upgraded to 5.1 + version, and we need to override the react-scripts to use typescript version 5 for work around. Do you have proper guideline on a proper solution for this matter?
